### ISSUE TITLE
Expose the entire context

### ISSUE THREAD

**Issue by FizzBuzz791** [thchia/react-oidc#5]
Working through the [React Context API](https://reactjs.org/docs/context.html) it is standard to assign the full context (Consumer and Provider) to `this.contextType` and then the context can be subsequently accessed via `this.context`. 

This package currently only exposes the Consumer, so it's not possible to do the above.

**Comment by thchia** [comment]
Thanks for the suggestion! Yes it seems I overlooked this way of accessing Context, therefore only supporting the Consumer component way.

I’ll get this out in the next few days, or feel free to open a PR!

**Comment by thchia** [comment]
Addressed in #6 and released in `v0.3.0`.

**Comment by eloekset** [comment]
How is this AuthenticatorContext made available? I'm trying to use it from a NavBar component, by writing things like:
```
const contextType = AuthenticatorContext;
// then to test whether the token is valid:
this.context.user.expired();
// and to call signin
this.context.userManager.signinRedirect();
```
But I get error message that this.context is undefined. When reading the React documentation for Context, it looks like that's how it's supposed to be used: https://reactjs.org/docs/context.html

**Comment by FizzBuzz791** [comment]
Have you created `AuthenticatorContext` first, using `React.createContext()` as per [https://reactjs.org/docs/context.html#classcontexttype](https://reactjs.org/docs/context.html#classcontexttype)? After that, you then need to do `MyClass.Context = AuthenticatorContext` in order to access it via `this.context`.

**Comment by thchia** [comment]
@eloekset As long as you have used `makeAuthenticator` as shown in the examples, then the Provider is already being rendered.

The next step is to declare the property on your class component before you can accessing using the `this.context` pattern, as mentioned above.

Alternatively you can use the render props pattern as demonstrated in the README [here](https://github.com/thchia/react-oidc#userdata-).

**Comment by eloekset** [comment]
Thanks, @thchia!

I saw that `makeAuthenticator` did that `React.createContext()` part, and expected the `this.context` to be available in my component by using the `const contextType = AuthenticatorContext` pattern, but it wasn't. I'm learning React, so I'm not able to understand why this won't work, but maybe there's a huge difference between `const`and `static`, as the React documentation uses. 

However, I've succeeded in getting access to the context by using the `<UserData.Consumer>` component now. If only the `context.user.expired()` function wasn't undefined, I've been able to render the user's name and a signout button or a signin button based on the expired value.

![image](https://user-images.githubusercontent.com/791721/56600869-04e72f80-65fa-11e9-9e67-269c02c1a91b.png)

Any idea why it's undefined? Or how I can be able to check if the current token is valid at this point? Is it perhaps not supposed to be a function, but rather a boolean value that happens to be either `false` or `undefined` if the token is not expired?
